File: sources/agency-notifier/src/main/java/fr/unice/polytech/si5/al/e/agencynotifier/interfaces/Notify.java
package fr.unice.polytech.si5.al.e.agencynotifier.interfaces;

import fr.unice.polytech.si5.al.e.model.*;

import javax.ejb.Local;

@Local
public interface Notify {

    void notifyContractReport(ContractSubscription contractSubscription, ItineraryStatus status);
    void notifyContractReport(Customer customer, Contract contract, Travel travel);
    void notifyContractRegister( ContractSubscription newSubscribtion);
}

File: blablamoveService/typescript/main/src/components/agency-notifier/Notify.ts
import {Subscribe} from "../../entity/Subscription/Subscribe";
import { ItineraryStatus } from "../../entity/ItineraryStatus";
import { Customer } from "../../entity/customer/Customer";
import { Contract } from "../../entity/contract/Contract";
import { Travel } from "../../entity/travel/Travel";

export interface Notify {

    notifyContractReport( subscription: Subscribe, status:ItineraryStatus): void
    notifyContractReportCustomer( customer: Customer, contract:Contract, travel: Travel): void
    notifyContractRegister( newSubscribtion: Subscribe):void
}